aea/aea_builder.py

```diff
@@ -604,6 +606,42 @@ def build(self, connection_ids: Optional[Collection[PublicId]] = None) -> AEA:
         self._load_and_add_skills(aea.context)
         return aea
 
+    # TODO: remove and replace with a clean approach (~noise based crypto module or similar)
+    def _update_identity(
```

Comment:
Note, @marcofavorito and @lrahmani this is a hack which only affects the case when the p2p_noise connection is used.

The problem with this connection is that its public key needs to be set as the agents default address. Currently, we do not support noise crypto and we are not sure we do want to. So this is a hack until a better solution is found.